quickwit-metastore/src/split_metadata.rs

```diff
@@ -0,0 +1,208 @@
+// Copyright (C) 2021 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::BTreeSet;
+use std::fmt;
+use std::ops::{Range, RangeInclusive};
+use std::str::FromStr;
+
+use chrono::Utc;
+use serde::{Deserialize, Serialize};
+
+/// Carries split and bundle offsets for single read metadata.
```

Comment:
by `for single read metadata`, do you mean that we should not mutate it?

Reply:
```suggestion
/// Carries split metadata and footer offsets. The footer offsets
/// make it possible to download the footer in a single call to `.get_slice(...)`.
```